specification/compute/resource-manager/Microsoft.Compute/preview/2021-06-01-preview/diagnostic.json

```diff
@@ -0,0 +1,394 @@
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ComputeDiagnosticBase"
+          },
+          "description": "The collection of available Compute diagnostics returned by the listing operation."
+        },
+        "nextLink": {
+          "type": "string",
+          "description": "The continuation token."
+        }
+      }
+    },
+    "ComputeDiagnosticBase": {
+      "description": "Contains metadata of a diagnostic type",
+      "type": "object",
+      "properties": {
+        "id": {
+          "description": "Identifier for the type of diagnostic",
+          "type": "string"
+        },
+        "name": {
+          "description": "Friendly name for the type of diagnostic",
+          "type": "string"
+        },
+        "type": {
+          "description": "Type of the diagnostic",
+          "type": "string"
+        },
+        "properties": {
+          "$ref": "#/definitions/DiagnosticProperties"
+        }
+      }
+    },
+    "DiagnosticProperties": {
+      "description": "Contains additional properties of a diagnostic",
+      "type": "object",
+      "properties": {
+        "supportedResourceTypes": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "Describes what are the supported resource types for a diagnostic."
+        }
+      }
+    },
+    "RunDiskInspectionInput": {
+      "type": "object",
+      "properties": {
+        "resourceId": {
+          "description": "Qualified name of the resource.",
+          "type": "string"
+        },
+        "manifest": {
+          "description": "Name of manifest in order to trigger Disk Inspection.",
+          "type": "string"
+        },
+        "uploadSasUri": {
+          "description": "SAS uri to the blob where results will be uploaded.",
+          "type": "string",
+          "x-ms-secret": true
+        }
+      },
+      "required": [
+        "resourceId",
+        "manifest",
+        "uploadSasUri"
+      ],
+      "description": "Data used for requesting a Disk Inspection execution."
+    },
+    "RunDiskInspectionAsyncOperationResult": {
+      "type": "object",
+      "properties": {
+        "result": {
+          "type": "string"
+        },
+        "resultStatus": {
+          "type": "string",
+          "enum": [
+            "Success",
+            "Failed"
+          ],
+          "x-ms-enum": {
+            "name": "resultStatus",
+            "modelAsString": true
+          }
+        },
+        "errorDetail": {
+          "$ref": "#/definitions/ErrorDetail"
+        },
+        "createdUTC": {
+          "type": "string",
+          "format": "date-time",
+          "description": "The time when the disk inspection was completed."
+        }
+      },
+      "description": "Api output result when disk inspection result is completed."
+    },
+    "ErrorResponse": {
+      "type": "object",
+      "x-ms-external": true,
```

Comment:
Hi @ansahdev Can you clarify why this is marked as x-ms-external:true? This prevents the object from being generated in the SDK which is resulting in SDK solution errors.

Reply:
I could have referred this from an existing RP json. What is the mitigation for the errors? Should this property be removed or updated with some additional information? Please advise.

Reply:
Removed the property.